### ISSUE TITLE
feature: core.typed annotations

### ISSUE THREAD

**Issue by lvh** [dakrone/cheshire#82]
Would this be welcomed if I started contributing them?

**Comment by michaelklishin** [comment]
If this will make Cheshire depend on `core.typed`, then I'd vote against it. If this can be made entirely optional, why not.

**Comment by dakrone** [comment]
I agree with Michael, if optional, yeah, if it added the dependency then probably not.

**Comment by lvh** [comment]
Would it be okay if it's a dev-time only dependency? `core.typed` is just a compile-time checker; it shouldn't show up e.g. production uberjars.

**Comment by dakrone** [comment]
@lvh I think a dev-time only dep is okay with me
